### ISSUE TITLE
Multiple definition of `max_packet_size` when building with GCC 10

### ISSUE THREAD

**Issue by bradfordb-vmware** [greenplum-db/gpdb#11772]
### Greenplum version or build

```console
$ ./getversion --short
7.0.0-alpha.0+dev.14504.g555b9c18a3
```

### OS version and uname -a

- Debian Bullseye **(`debian:testing@sha256:52351528547bc933a54b44714e300f9d5e72208774678b33935ee5ffc7af803a`)**
- Ubuntu 21.04 (`ubuntu:21.04@sha256:0cd296c77ae19141240957d2a57f9a23c95b05f7c930fe5e509a4880dbe0ac13`)

### autoconf options used ( config.status --config )

```console
$ ./config.status --config
`--with-gssapi`
```

### Installation information ( pg_config )

N/A

### Expected behavior

Greenplum builds successfully

### Actual behavior

```console
$ make -j4
...
/usr/bin/ld: libpq/fe-secure-gssapi.o:(.bss+0x0): multiple definition of `max_packet_size'; libpq/be-secure-gssapi.o:(.bss+0x0): first defined here
collect2: error: ld returned 1 exit status
```

### Step to reproduce the behavior

```bash
apt update
DEBIAN_FRONTEND=noninteractive apt install -q -y \
bison \
flex \
g++-9 \
g++-10 \
gcc-9 \
gcc-10 \
libapr1-dev \
libbz2-dev \
libcurl4-openssl-dev \
libevent-dev \
libkrb5-dev \
libreadline-dev \
libxerces-c-dev \
libzstd-dev \
make \
pkg-config \
python3 \
python3-dev

# this works
CC=gcc-9 CXX=g++-9 ./configure --with-gssapi
make -j4

# this does not
CC=gcc-10 CXX=g++-10 ./configure --with-gssapi
make -j4
```

**Comment by d** [comment]
I bet 1 BTC that you'll fail with a similar error if you use `CFLAGS=-fcommon` for GCC 9, can you try that?

**Comment by bradfordb-vmware** [comment]
`CC=gcc-9 CXX=g++-9 CFLAGS=-fcommon ./configure --with-gssapi` works.

**Comment by bradfordb-vmware** [comment]
paired w/ @d and the flag we wanted was `-fno-common`. With this in place, we see the error when compiling with gcc 9.

**Comment by bradfordb-vmware** [comment]
It looks like this may have been a bug introduced in Postgres 12 [commit](https://github.com/postgres/postgres/commit/b0b39f72b9904bcb80f97b35837ccff1578aa4b8) that was fixed in Postgres 13 [commit](https://github.com/postgres/postgres/commit/2c0cdc8183654c090c9a1e2f1b5e96ba4634e16a) and back-patched to 12 [commit](https://github.com/postgres/postgres/commit/fde155424f6779b009f8f01a71af8dd6176a0c81).

Should we look at back-patching this as well?

**Comment by asimrp** [comment]
We need to merge REL_12_STABLE branch from upstream into Greenplum 7 before releasing 7.0.  If a specific commit(s) is needed to reduce your hassle, please feel free to cherry-pick it.  Else, it will be brought it by the REL_12_STABLE merge.  FYI, @wuj21 @wenwangBJ 

Good catch, by the way!

**Comment by bradfordb-vmware** [comment]
> If a specific commit(s) is needed to reduce your hassle, please feel free to cherry-pick it. Else, it will be brought it by the REL_12_STABLE merge.

I can work around this for the time being by either omitting kerberos support (i.e., `--without-gssapi`) or setting `CFLAGS=-fcommon`. I'll leave it to merge experts to bring the fix in when REL_12_STABLE is merged prior to releasing GPDB 7.0.0.

@asimrp Should I leave this issue open until then or should we consider this closed?

**Comment by adam8157** [comment]
I filed this issue months ago, https://github.com/greenplum-db/gpdb/issues/11551

This is annoying to me, will backport the patch.

**Comment by adam8157** [comment]
Fixed by #11832
